### ISSUE TITLE
Better monitoring of `npm version` progress

### ISSUE THREAD

**Issue by jamestalmage** [bjoerge/cut-release#12]
Since the `npm version` command can now [run an entire build process](https://github.com/npm/npm/issues/8620), it would be great to see a bit more detail about the progress of that command.

I don't know of an easy way to actually accomplish that that isn't hackie and brittle, so this may be a no go.

Just food for thought.

**Comment by bjoerge** [comment]
Interesting. The output from the subprocess that calls `npm version` is currently just forwarded to stdout. Would you like more fine grained progress info, like `=> Running preversion`, `=> Running version`, etc?

**Comment by jamestalmage** [comment]
> The output from the subprocess that calls npm version is currently just forwarded to stdout

Yes, but it buffers until the process finishes. With a long build process, there is no indicator if you are making progress or hung.

> => Running preversion, => Running version

That is what I was thinking (initially).

It may be possible to simply grep the output stream of `npm version` for the start of each stage. I am not sure how brittle that would be, but I am sure we could have it degrade gracefully in the event npm changes the specifics of the output.

Perhaps also a "dot" progress indicator (each dot representing a line printed to stdout by the child process).

```
Running preversion: ........
Running version: ....
```

If the process exits with a nonzero error code, we definitely should dump stdout, and stderr to the console.

We should probably also include a `--verbose` flag that dumps stdout and stderr in real time.

**Comment by jamestalmage** [comment]
It seems the boundary between stages is well defined 

> &gt; **package-name**@**0.2.0** preversion **/path/to/basedir**
> &gt; **contents_of_preversion_script**
> ...
> ... _preversion_stdout_
> ...
> &gt; **package-name**@**0.2.0** version **/path/to/basedir**
> &gt; **contents_of_version_script**
> ...
> ... _version_stdout_
> ...

**Comment by jamestalmage** [comment]
Also, in `--verbose` mode, we might as well pipe `stdin` into the process (in case their build has prompts).
